src/api/modules/utils.ts

```diff
@@ -129,81 +128,94 @@ export const appIdFromRegistry = (app: string, majorLocator: string) => {
+  }
+  if (policies && policies.length > 0) {
+    return billingInfoFromPolicy(policies[0])
+  }
+}
+
+export function optionsFormatter({ plans, policies, type, free }: BillingOptions, licenseURL?: string) {
   /** TODO: Eliminate the need for this stray, single `cli-table2` dependency */
   const table = new Table({
     head: [{ content: chalk.cyan.bold('Billing Options'), colSpan: 2, hAlign: 'center' }],
     chars: { 'top-mid': '─', 'bottom-mid': '─', 'mid-mid': '─', middle: ' ' },
   })
-
-  if (billingOptions.free) {
-    table.push([{ content: chalk.green('This app is free'), colSpan: 2, hAlign: 'center' }])
+  const billingInfo = buildBillingInfo(plans, policies)
+  if (isFreeApp(type, free) || !billingInfo) {
+    table.push([{ content: chalk.green('Free app'), colSpan: 2, hAlign: 'center' }])
   } else {
-    table.push([
-      { content: 'Plan', hAlign: 'center' },
-      { content: 'Values', hAlign: 'center' },
-    ])
-
-    billingOptions.policies.forEach(policy => {
-      let rowCount = 0
-      const itemsArray = []
-
-      policy.billing.items.forEach(i => {
-        if (i.fixed) {
-          itemsArray.push([{ content: `${i.fixed} ${i.itemCurrency}`, hAlign: 'center', vAlign: 'center' }])
-          rowCount++
-        } else if (i.calculatedByMetricUnit) {
-          if (i.calculatedByMetricUnit.minChargeValue) {
-            itemsArray.push([`Minimum charge: ${i.calculatedByMetricUnit.minChargeValue} ${i.itemCurrency}`])
-            rowCount++
-          }
-
-          let rangesStr = ''
-          i.calculatedByMetricUnit.ranges.forEach(r => {
-            if (r.inclusiveTo) {
-              rangesStr += `${r.multiplier} ${i.itemCurrency}/${i.calculatedByMetricUnit.metricName} (${r.exclusiveFrom} to ${r.inclusiveTo})`
-              rangesStr += '\nor\n'
-            } else {
-              rangesStr += `${r.multiplier} ${i.itemCurrency}/${i.calculatedByMetricUnit.metricName} (over ${r.exclusiveFrom})`
-            }
-          })
-
-          rowCount++
-          itemsArray.push([{ content: rangesStr, hAlign: 'center', vAlign: 'center' }])
-        }
-        itemsArray.push([{ content: '+', hAlign: 'center' }])
-        rowCount++
-      })
-
-      itemsArray.pop()
-      rowCount--
-
-      table.push(
-        [
-          {
-            content: `${chalk.yellow(policy.plan)}\n(Charged montlhy)`,
-            rowSpan: rowCount,
-            colSpan: 1,
-            vAlign: 'center',
-            hAlign: 'center',
-          },
-          itemsArray[0][0],
-        ],
-        ...itemsArray.slice(1)
-      )
-      table.push([
-        {
-          content: `The monthly amount will be charged in ${chalk.red(policy.currency)}`,
-          colSpan: 2,
-          hAlign: 'center',
-        },
-      ])
-    })
+    chalkBillingTable(table, billingInfo)
+  }
+  if (licenseURL && licenseURL.length > 0) {
+    table.push([{ content: chalk.yellow.bold('App Terms of Service: ') }, { content: licenseURL }])
```

Comment:
Same here

Reply:
Done in 1a1f01e7e948ad1dcbe61c6a7cc761f9a28c6dd4, thanks!